test/Datadog.Trace.ClrProfiler.IntegrationTests/ProcessHelper.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    /// <summary>
+    /// Drains the standard and error output of a process
+    /// </summary>
+    internal class ProcessHelper : IDisposable
+    {
+        private readonly ManualResetEventSlim _errorMutex = new();
+        private readonly ManualResetEventSlim _outputMutex = new();
+        private readonly StringBuilder _outputBuffer = new();
+        private readonly StringBuilder _errorBuffer = new();
+
+        public ProcessHelper(Process process)
+        {
+            process.OutputDataReceived += (_, e) => DrainOutput(e, _outputBuffer, _outputMutex);
+            process.ErrorDataReceived += (_, e) => DrainOutput(e, _errorBuffer, _errorMutex);
+
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+        }
+
+        public string StandardOutput => _outputBuffer.ToString();
+
+        public string ErrorOutput => _errorBuffer.ToString();
+
+        public bool Drain(int timeout = Timeout.Infinite)
+        {
+            return _outputMutex.Wait(timeout / 2) && _errorMutex.Wait(timeout / 2);
```

Comment:
I think we should have a case when timeout is Timeout.Infinite, feels wrong the `timeout/2` in that case.